packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -383,6 +384,14 @@ abstract class BucketBase extends Resource implements IBucket {
    */
   protected abstract disallowPublicAccess?: boolean;
 
+  private readonly skipPutObjectAclInGrantWrite: boolean | undefined;
```

Comment:
How does `isEnabled` return `undefined`? Dafuq?

Also, couldn't we just inline the feature flag check into `putActions`?

Reply:
Yeah, fair enough ;p. I _think_ in a previous version I read the field-backed expression twice, hence why I introduced it, but then I refactored the code to only do it once, and that made this field kinda redundant... Removed!